Build failure on MacOS Sequoia

```
$ stack install

Warning: nix (Nix) 2.23.3 is on the PATH (at /nix/var/nix/profiles/default/bin/nix) but Stack's Nix
         integration is disabled. To mute this message in future, set notify-if-nix-on-path: false in
         Stack's configuration.

kmonad-0.4.3: unregistering (flags changed from
["-fdext","-f-kext","--extra-include-dirs=/Users/sooheon/code/kmonad/c_src/mac/Karabiner-VirtualHIDDevice/dist/include"]
to ["-f-dext","-f-kext"])
kmonad> configure (lib + exe)
Configuring kmonad-0.4.3...
kmonad> build (lib + exe) with ghc-9.6.6
Preprocessing library for kmonad-0.4.3..
Building library for kmonad-0.4.3..
[ 1 of 35] Compiling KMonad.Prelude.Imports

...

[2 of 2] Linking .stack-work/dist/aarch64-osx/ghc-9.6.6/build/kmonad/kmonad [Objects changed]
ld: warning: search path '/usr/local/gfortran/lib' not found
Undefined symbols for architecture arm64:
  "_grab_kb", referenced from:
       in libHSkmonad-0.4.3-1fZVv7QycZ97pjeP2S2kqT.a[34](IOKitSource.o)
       in libHSkmonad-0.4.3-1fZVv7QycZ97pjeP2S2kqT.a[34](IOKitSource.o)
  "_release_kb", referenced from:
       in libHSkmonad-0.4.3-1fZVv7QycZ97pjeP2S2kqT.a[34](IOKitSource.o)
  "_send_key", referenced from:
       in libHSkmonad-0.4.3-1fZVv7QycZ97pjeP2S2kqT.a[35](KextSink.o)
  "_wait_key", referenced from:
       in libHSkmonad-0.4.3-1fZVv7QycZ97pjeP2S2kqT.a[34](IOKitSource.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ghc-9.6.6: `gcc' failed in phase `Linker'. (Exit code: 1)

Error: [S-7282]
       Stack failed to execute the build plan.

       While executing the build plan, Stack encountered the error:

       [S-7011]
       While building package kmonad-0.4.3 (scroll up to its section to see the error) using:
       /Users/sooheon/.stack/setup-exe-cache/aarch64-osx/Cabal-simple_DY68M0FN_3.10.3.0_ghc-9.6.6 --verbose=1 --builddir=.stack-work/dist/aarch64-osx/ghc-9.6.6 build lib:kmonad exe:kmonad --ghc-options " -fdiagnostics-color=always"
       Process exited with code: ExitFailure 1
```

--- a/c_src/mac/kext.cpp
+++ b/c_src/mac/kext.cpp
@@ -41,7 +41,7 @@ int exit_sink() {
 int init_sink() {
     kern_return_t kr;
     connect = IO_OBJECT_NULL;
-    service = IOServiceGetMatchingService(kIOMasterPortDefault, IOServiceNameMatching(pqrs::karabiner_virtual_hid_device::get_virtual_hid_root_name()));
+    service = IOServiceGetMatchingService(kIOMainPortDefault, IOServiceNameMatching(pqrs::karabiner_virtual_hid_device::get_virtual_hid_root_name()));
     if (!service) {
         print_iokit_error("IOServiceGetMatchingService");
         return 1;

--- a/c_src/mac/keyio_mac.hpp
+++ b/c_src/mac/keyio_mac.hpp
@@ -9,7 +9,8 @@
 #include <AvailabilityMacros.h>
 
 /* The name was changed from "Master" to "Main" in Apple SDK 12.0 (Monterey) */
-#if !defined (MAC_OS_X_VERSION_12_0) || (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_12_0)
+/* MAC_OS_X_VERSION_12_0 does not exist :( */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 120000
     #define kIOMainPortDefault kIOMasterPortDefault
 #endif
 

--- a/doc/installation.md
+++ b/doc/installation.md
@@ -283,22 +283,18 @@ and install `stack`
   $ brew install haskell-stack
 ```
 
-Then build kmonad with `stack`. If you are building against the kext, run:
+Then build kmonad with `stack` and install it to `~/.local/bin/`
+(which you may want to ensure is on your `PATH`).
+If you are building against the kext, run:
 
 ``` console
-  $ stack build --flag kmonad:kext --extra-include-dirs=c_src/mac/Karabiner-VirtualHIDDevice/dist/include
+  $ stack install --flag kmonad:kext --extra-include-dirs=c_src/mac/Karabiner-VirtualHIDDevice/dist/include
 ```
 
 If you are building against the dext, run
 
 ``` console
-  $ stack build --flag kmonad:dext --extra-include-dirs=c_src/mac/Karabiner-DriverKit-VirtualHIDDevice/include/pqrs/karabiner/driverkit:c_src/mac/Karabiner-DriverKit-VirtualHIDDevice/src/Client/vendor/include
-```
-
-Finally, run the following to place the executable `kmonad` in `~/.local/bin/` (which you may want to ensure is on your `PATH`)
-
-``` console
-  $ stack install
+  $ stack install --flag kmonad:dext --extra-include-dirs=c_src/mac/Karabiner-DriverKit-VirtualHIDDevice/include/pqrs/karabiner/driverkit:c_src/mac/Karabiner-DriverKit-VirtualHIDDevice/src/Client/vendor/include
 ```
 
 #### Giving kmonad additional permissions

